src/main/java/team/themoment/hellogsmv3/domain/oneseo/service/DownloadExcelService.java

```diff
@@ -0,0 +1,172 @@
+            Sheet sheet = sheets.get(i);
+            List<List<String>> sheetData = allSheetData.get(i);
+            populateSheet(sheet, sheetData);
+        }
+
+        return workbook;
+    }
+
+    private List<Sheet> createSheets(Workbook workbook) {
+        return List.of(
+                workbook.createSheet("일반전형"),
+                workbook.createSheet("사회전형"),
+                workbook.createSheet("특별전형"),
+                workbook.createSheet("불합격")
+        );
+    }
+
+    private void populateSheet(Sheet sheet, List<List<String>> data) {
+        int rowCount = 0;
+        Row headerRow = sheet.createRow(rowCount++);
+        addDataToRow(headerRow, HEADER_NAMES);
+
+        for (List<String> rowData : data) {
+            Row row = sheet.createRow(rowCount++);
+            addDataToRow(row, rowData);
+        }
+    }
+
+    private void addDataToRow(Row row, List<String> dataList) {
+        int cellCount = 0;
+        for (String data : dataList) {
+            Cell cell = row.createCell(cellCount++);
+            cell.setCellValue(data);
+        }
+    }
+
+    private List<List<List<String>>> getAllSheetData() {
+        return List.of(
+                getOneseoData(Screening.GENERAL),
+                getOneseoData(Screening.SPECIAL),
+                getCombinedExtraScreeningData(),
+                getFallenData()
+        );
+    }
+
+    private List<List<String>> getOneseoData(Screening screening) {
+        List<Oneseo> oneseoList = oneseoRepository.findAllByAppliedScreeningAndFinalSubmittedYn(screening, YesNo.YES);
+        return oneseoToExcelDataList(oneseoList);
+    }
+
+    private List<List<String>> getCombinedExtraScreeningData() {
+        List<Oneseo> extraOneseoList = Stream.concat(
+                oneseoRepository.findAllByAppliedScreeningAndFinalSubmittedYn(Screening.EXTRA_VETERANS, YES).stream(),
+                oneseoRepository.findAllByAppliedScreeningAndFinalSubmittedYn(Screening.EXTRA_ADMISSION, YES).stream()
+        ).collect(Collectors.toList());
+
+        return oneseoToExcelDataList(extraOneseoList);
+    }
+
+    private List<List<String>> getFallenData() {
+        List<Oneseo> fallenOneseo = entranceTestResultRepository
+                .findAllByFirstTestPassYnOrSecondTestPassYnAndOneseoFinalSubmittedYn(NO, NO, YES).stream()
+                .map(EntranceTestResult::getOneseo)
+                .collect(Collectors.toList());
+
+        return oneseoToExcelDataList(fallenOneseo);
+    }
+
+    private List<List<String>> oneseoToExcelDataList(List<Oneseo> oneseoList) {
+        List<List<String>> sheetData = new ArrayList<>();
+        int index = 1;
+
+        for (Oneseo oneseo : oneseoList) {
+            EntranceTestResult entranceTestResult = entranceTestResultRepository.findEntranceTestResultByOneseo(oneseo)
+                    .orElseThrow(() -> new ExpectedException("해당 원서로 입학 시험 결과를 찾을 수 없습니다. oneseo ID: " + oneseo.getId(), HttpStatus.NOT_FOUND));
```

Comment:
```suggestion
            EntranceTestResult entranceTestResult = entranceTestResultRepository.findByOneseo(oneseo);
```

비즈니스 로직상 최종제출된 원서는 EntranceTestResult 엔티티와 식별관계로 존재하도록 되어있어 위처럼 Optional로 조회하지 않아도 될듯 합니다.

Reply:
옵셔널 제거하였습니다.
[602aee4](https://github.com/themoment-team/hellogsm-server-24/pull/93/commits/602aee42d8e54842576614255b30031e9edbe755)